server/src/test/java/org/elasticsearch/plugins/PluginInfoTests.java

```diff
@@ -84,6 +84,23 @@ public void testReadFromPropertiesElasticsearchVersionMissing() throws Exception
         assertThat(e.getMessage(), containsString("[elasticsearch.version] is missing"));
     }
 
+    public void testReadFromPropertiesElasticsearchVersionEmpty() throws Exception {
+        Path pluginDir = createTempDir().resolve("fake-plugin");
+        PluginTestUtil.writePluginProperties(
+            pluginDir,
+            "description",
+            "fake desc",
+            "name",
+            "my_plugin",
+            "version",
+            "1.0",
+            "elasticsearch.version",
+            ""
```

Comment:
Since we are using the Strings util, should we test for non zero length but still empty?